### PR TITLE
feat: linting rules for SDK

### PR DESCRIPTION
**Pull request by ssbarbee**
lets have nice rules for this library 🙏🏻

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,15 +13,18 @@
   "scripts": {
     "prepare": "is-ci || husky install",
     "build": "tsc -p tsconfig.build.json",
+    "prebuild-release": "yarn code-quality:check",
```
Comment (avniculae) [inline package.json:16]: is this used anywhere?

Comment (avniculae) [inline package.json:16]: is this used anywhere?

File: src/typechain/index.ts [hunk 2]
```diff
@@ -1,25 +1,24 @@
 /* Autogenerated file. Do not edit manually. */
```
[the remark this answers is not in the document]
Comment (0xZenus) [reply src/typechain/index.ts:1]: I agree. ``typechain`` is auto-generated in the smart contract repositories.

File: package.json (hunk 1 above)
Comment (ssbarbee) [reply package.json:16]: yes. Feel free to read: https://docs.npmjs.com/cli/v9/using-npm/scripts#pre--post-scripts to get better knowledge. But in short whenever you run `yarn build-release` a `prebuild-release` is run

File: src/typechain/index.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (ssbarbee) [reply src/typechain/index.ts]: noted and reverted!

**Review by avniculae — APPROVED** [review]

**Pull request merged**